recipes-extended/xen/files/efi-require-shim.patch

```diff
@@ -0,0 +1,30 @@
+From c2b3574915adf8ef9d4863e6617b8cfb5d861c3c Mon Sep 17 00:00:00 2001
+From: Tamas K Lengyel <tamas@tklengyel.com>
+Date: Sun, 25 Feb 2018 13:17:01 -0700
+Subject: [PATCH] Require the presence of the shim to boot under UEFI
+
+---
+ xen/common/efi/boot.c | 7 +++++++
+ 1 file changed, 7 insertions(+)
+
+diff --git a/xen/common/efi/boot.c b/xen/common/efi/boot.c
+index 8115dd57d9..2c1be122aa 100644
+--- a/xen/common/efi/boot.c
++++ b/xen/common/efi/boot.c
+@@ -1237,6 +1237,13 @@ efi_start(EFI_HANDLE ImageHandle, EFI_SYSTEM_TABLE *SystemTable)
+ 
+         efi_bs->LocateProtocol(&shim_lock_guid, NULL, (void **)&shim_lock);
+ 
++        if ( !shim_lock )
++        {
++            PrintStr(L"No shim found, booting xen.efi directly with OpenXT is not supported.");
++            efi_bs->Stall(10000000);
++            blexit(L"No shim found");
++        }
++
+         if ( shim_lock &&
+             (status = shim_lock->Measure(cfg.ptr, cfg.size, 4)) != EFI_SUCCESS )
```

Comment:
Would it make sense remove all the following "shim_lock &&" with the exit above?  Can this be squashed into the shim-support-for-shim-lock-measure.patch?

Reply:
It is kept in this form because the patch that errors out when the shim protocol is not found is not something that can be ever upstreamed into Xen, whereas calling Measure could be once my PR is merged into the shim, in which case they will need to be gated on shim_lock being defined.

Reply:
Okay.